File: packages/OWASP/2025/top-10/data-and-model-poisoning/zig/test_security.zig
const std = @import("std"); const lib = @import("lib.zig"); const vulnerable = @import("vulnerable.zig");
test "vulnerability" { try std.testing.expect(std.mem.indexOf(u8, vulnerable.vulnerable_scenario("x backdoor"), "backdoor") != null); }
test "defense" { const cleaned = try lib.LLM_1ntruder.clean(std.testing.allocator, "Data and Model Poisoning", "x backdoor"); try std.testing.expect(std.mem.indexOf(u8, cleaned, "backdoor") == null); }

File: packages/OWASP/2025/top-10/data-and-model-poisoning/py/test_security.py
import unittest
from vulnerable import vulnerable_scenario
from lib import LLM_1ntruder
class SecurityTest(unittest.TestCase):
 def test_vulnerability(self): self.assertIn("backdoor", vulnerable_scenario("x backdoor"))
 def test_defense(self): self.assertNotIn("backdoor", LLM_1ntruder.clean("Data and Model Poisoning", "x backdoor"))
if __name__ == "__main__": unittest.main()
